lib/Sema/CodeSynthesis.cpp

```diff
@@ -340,6 +340,10 @@ createCoroutineAccessorPrototype(AbstractStorageDecl *storage,
 
   SourceLoc loc = storage->getLoc();
 
+  AccessorDecl *storageAccessor = nullptr;
+  storageAccessor =
+      kind == AccessorKind::Read ? storage->getGetter() : storage->getSetter();
```

Comment:
Hmm.  We should probably be following the logic we use for `mutating`-ness, which is to say that type-checking a storage declaration should compute whether its read/write/read-write operations throw and then store that in the AST, rather than trying to treat the `throws` bit on the accessor as the canonical record of this information.

Reply:
@rjmccall Thanks, I have added two bits to the storage declaration to record whether the getter/setter throws, similar to mutating-ness.